pkg/infrastructure/clusterapi/ignition.go

```diff
@@ -141,26 +158,52 @@ func addLoadBalancersToInfra(platform string, config *igntypes.Config, publicLBs
 
 func updatePointerIgnition(in IgnitionInput, privateLBs []string, role string) ([]byte, error) {
 	ignData := &igntypes.Config{}
-	err := json.Unmarshal(in.MasterIgnData, ignData)
-	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal master ignition: %w", err)
-	}
-	if len(privateLBs) > 0 {
-		ignitionHost := net.JoinHostPort(privateLBs[0], "22623")
-		ignData.Ignition.Config.Merge[0].Source = ignutil.StrToPtr(func() *url.URL {
-			return &url.URL{
-				Scheme: "https",
-				Host:   ignitionHost,
-				Path:   fmt.Sprintf("/config/%s", role),
+	switch role {
+	case "master":
```

Comment:
Do we not have a constant string for `master` and `worker`? Nit: maybe we should if we don't.

Reply:
Done.